### PR TITLE
Qt/NetPlay: Show feedback for index adding

### PR DESCRIPTION
**Pull request by spycrab**
Shows a message when:
* There has been an error while adding a session to the index
* Adding the session has succeeded
* The game has been dropped from the index

### CODE REVIEW

File: Source/Core/Core/NetPlayServer.cpp [hunk 1]
```diff
@@ -183,6 +183,9 @@ void NetPlayServer::SetupIndex()
 
   if (m_traversal_client)
   {
+    if (m_traversal_client->GetState() == TraversalClient::Connected)
```
Comment (Techjar) [inline Source/Core/Core/NetPlayServer.cpp]: Surely this should be `!=`.

File: Source/Core/UICommon/NetPlayIndex.cpp [hunk 2]
```diff
@@ -318,3 +318,8 @@ const std::string& NetPlayIndex::GetLastError() const
 {
   return m_last_error;
 }
+
+bool NetPlayIndex::HasActiveSession() const
+{
+  return !m_secret.empty();
```
Comment (Techjar) [inline Source/Core/UICommon/NetPlayIndex.cpp]: This doesn't properly indicate that the session is still active, as the notification loop can exit due to failure and `m_secret` will still be non-empty. Should probably make it empty when the notification loop returns.

File: Source/Core/Core/NetPlayServer.cpp [hunk 3]
```diff
@@ -221,11 +225,22 @@ void NetPlayServer::ThreadFunc()
...
 
-      m_update_pings = false;
+        m_update_pings = false;
+      }
+      else
+      {
+        if (m_dialog != nullptr)
+        {
+          m_dialog->OnIndexRefreshFailed(m_index.GetLastError());
+          Config::SetCurrent(Config::NETPLAY_USE_INDEX, false);
```
Comment (Techjar) [inline Source/Core/Core/NetPlayServer.cpp]: This probably shouldn't be here. In fact this code path will execute every time it's called if the index is disabled or failed.

Comment (spycrab) [reply Source/Core/Core/NetPlayServer.cpp]: Not the cause of the massive slowdown though.

Comment (Techjar) [reply Source/Core/Core/NetPlayServer.cpp]: Yeah I know, that was totally unrelated, but this still spams the chat.

File: Source/Core/Core/NetPlayServer.cpp [hunk 4]
```diff
@@ -201,7 +204,13 @@ void NetPlayServer::SetupIndex()
 
   session.EncryptID(Config::Get(Config::NETPLAY_INDEX_PASSWORD));
 
-  m_index.Add(session);
+  if (m_dialog != nullptr)
+    m_dialog->OnIndexAdded(m_index.Add(session), m_index.GetLastError());
+
+  // m_index.SetErrorCallback([this] {
```
Comment (Techjar) [inline Source/Core/Core/NetPlayServer.cpp]: What is this?

**Pull request merged**